### ISSUE TITLE
Doesn't work with the 433 version of E220

### ISSUE THREAD

**Issue by sn0ren** [nonik0/CardputerLoRaChat#5]
It doesn't seem to work with my 433 MHz version of the E220 module

**Comment by nonik0** [comment]
Sorry it's not working. I only have the module for my region so I am unable to debug. I could try to help if you provide some more details? Do you have 2 Cardputers with this same app installed trying to communicate?

**Comment by sn0ren** [comment]
Yeah I have 2 Cardputers with 2 modules and the same app installed. No message comes through. It works fine using ESP chat. I guess it's because the module library is region specific?

**Comment by nonik0** [comment]
It's been a while so I took another look at the E220 library and I don't believe it should have to do with that, at least AFAICT.

It's definitely possible that I maybe have a bug with the LoRa config not actually being written despite a green OK, so I would do extra attempts/debugging around that to make sure the LoRa modules are really configured the same/default and the switches are set properly for both writing config and RX/TX for radio. Or, if you've gotten the LoRa modules to talk in some other LoRa app, then it definitely would seem to be an issue with this app that would need some more debugging.

**Comment by Kunceton** [comment]
Hello nonik0,
Appreciate you developed this app.
I Just got a cardputer and 433MHz E220 module, but I met the same problem.
I’d like to know if you still working on this project? Will 433Mhz frequency be support in the future?
Thank you!

**Comment by gsgalezowski** [comment]
What version of the Lora card does this program work with? It doesn't work on 433MHz E220 at my place either.

**Comment by nonik0** [comment]
Without buying 2x of these modules to be able to repro and subsequently fix the issue, I am afraid I am not sure how to proceed since I can't see from the code where the issue is (it has to be the module config??).  Would be happy to fix if I was able to pinpoint issue, so any help is appreciated.

@Gitshaoxiang (author of E220 library)

**Comment by joakim** [comment]
I just bought a 433MHz E220 module for my Cardputer. Is there any way I can help solve this issue? I only have one set, but maybe that's enough to get some logs to figure out what's happening.

**Comment by joakim** [comment]
@nonik0 I see that you use the [M5-LoRa-E220-JP](https://github.com/m5stack/M5-LoRa-E220-JP) library. @Gitshaoxiang has also made a [M5-LoRa-E220](https://github.com/m5stack/M5-LoRa-E220) library, maybe that's the one to use for the 433Mhz module?

The code is almost the same, but M5-LoRa-E220 has an extra [`uart_config`](https://github.com/m5stack/M5-LoRa-E220/blob/main/src/M5_LoRa_E220.h#L13C13-L13C24) config item and different [typedefs](https://github.com/m5stack/M5-LoRa-E220/blob/main/src/M5_LoRa_E220_DEF.h).

There's also [an Arduino example](https://github.com/m5stack/M5-LoRa-E220/blob/69a2bdf350d3d60b84716f53f7e3a0fd5556808c/examples/P2P_TX_RX/P2P_TX_RX.ino).

**Comment by gsgalezowski** [comment]
Hi
I bought the recommended for the LoRaE220-JP program but not sure if it works. I don't know if I need to buy a LoRaE220-JP module for my other Cardputer to see if they communicate with each other using CardputerLoRaChat?
![IMG_20250112_111350049](https://github.com/user-attachments/assets/26e111ad-ed47-4ecd-97c6-cd803f0668f4)
![IMG_20250112_111406968](https://github.com/user-attachments/assets/a111abaf-363d-4daf-a8b2-e2c16a19226f)
I have attached pictures of the program running with the Lora920 module connected. Is it possible to determine from these pictures whether this module has been detected?

**Comment by joakim** [comment]
@gsgalezowski Are the bars of your signal strength indicator solid? Looks that way. Mine are hollow, and my Cardputer isn't connected to anything.

Edit: Never mind, I zoomed in and can see that yours are hollow too.

**Comment by gsgalezowski** [comment]
It seems that the signal level indicators look the same as with the lora 920 module disconnected. That is, it does not work for me even on this recommended module. With the 433 module connected it looks the same.

**Comment by joakim** [comment]
~Good news! I forked the repo, swapped out the M5-LoRa-E220-JP library with M5-LoRa-E220, built it, and it seems to work!~

~I got orange bars, and the status bar shows an arrow when I send a message. I don't have another device to test with, but the users tab did show one user with no name and RSSI -1. That was strange. After a reboot, I don't see the user anymore and my bars are no longer orange, but white and hollow. But it still seems to work, showing an arrow when I send a message.~

https://github.com/joakim/CardputerLoRaChat

Edit: Not good news 😕 It does exactly the same without the module connected. Seems like it only worked the first time when I got orange bars. So it's a promising start, but it's not working yet. Unfortunately, I only have one CardPuter to test with (I know, I should have bought two).

@nonik0 Any ideas why it would only work once?

**Comment by joakim** [comment]
I did some testing, and I did get orange bars one more time, and the Users Seen tab showed:

```
RSSI: -1, last seen: 42s
```

So that's something.

Edit: I've looked closer at the code, and turns out the bars only turn orange after receiving a LoRa frame, because that's when you get an RSSI. But -1 means something's wrong. I don't get any errors when monitoring through PlatformIO, and I don't have a second device to check whether TX actually works.

I might try to create one with an old E32-TTL-100 433Mhz module I have lying around, on a Raspberry Pi or something. I've studied nonik0's code, and it shouldn't be too hard to make a chat app on another platform that sends compatible LoRa frames. Assuming the two LoRa modules are capable of talking to eachother (E32-TTL-100 is based on SX1278).

**Comment by gsgalezowski** [comment]
Why is there a loss of coverage after 3 minutes? This applies to both the 920 and 433 versions.

**Comment by joakim** [comment]
Update: I hooked up two E32-TTL-100s on two old microcontrollers, and they are communicating just fine over LoRa. But they're not picking up anything from the E220 😕

So no, even if the firmware UI gives the impression that it works, and monitoring from PlatformIO shows no errors, it doesn't seem to be transmitting anything.
